python/lsst/consdb/pqserver.py

```diff
@@ -680,17 +701,32 @@ def insert_flexible_metadata(
         if dtype != type(value).__name__:
             raise BadValueException(f"{dtype} value", value)
 
+    has_multi_column_primary_keys = (
```

Comment:
The intent here was to change the actual APIs (or add new ones) to take `day_obs` and `seq_num` (possibly with `detector`) instead of `obs_id`.  I realize that adds even more complexity, but it's mostly translating those to `obs_id` and then calling this one.

Reply:
As I see it, this is a good step 1, and then if we deploy this and it looks good then we can move to changing the APIs as step 2.

Reply:
I wouldn't remove support for the old APIs (step 3), but doing this in two steps is fine.